src/Bicep.Core/Workspaces/SourceFileGroupingBuilder.cs

```diff
@@ -22,8 +29,11 @@ public class SourceFileGroupingBuilder
 
         private readonly Dictionary<Uri, ResultWithDiagnostic<ISourceFile>> fileResultByUri;
         // For each .bicep file key, a dictionary (keyed by 'module' or other syntax statement in the .bicep file) of the resolved URI for that syntax statement
-        private readonly ConcurrentDictionary<BicepSourceFile, Dictionary<IArtifactReferenceSyntax, Result<Uri, UriResolutionError>>> fileUriResultByArtifactReference;
-
+        private readonly ConcurrentDictionary<BicepSourceFile, Dictionary<IArtifactReferenceSyntax, Result<Uri, UriResolutionError>>> uriResultByBicepSourceFileByArtifactReferenceSyntax;
```

Comment:
Why do we need these to be concurrent? Do we ever call into this class in a multithreaded context?

Reply:
It appears that we were taking advantage of the the `GetOrAdd` signature which allows to initialize an entry in a single line but we can acomplish this using `TryGetValue` as well, its just a bit more verbose... I went ahead and changed it